src/core/load-fragment.c

```diff
@@ -3012,6 +3013,48 @@ int config_parse_cpu_quota(
         return 0;
 }
 
+int config_parse_cpu_quota_period(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        CGroupContext *c = data;
+        usec_t u;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                c->cpu_quota_period_usec = 0;
+                return 0;
+        }
+
+        r = parse_time(rvalue, &u, USEC_PER_MSEC);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Invalid CPU quota period '%s', ignoring.", rvalue);
+                return 0;
+        }
+
+        /* Valid range for period in cpu.max is [1ms, 1s]:
+         * https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/kernel/sched/core.c?h=v4.18-rc5#n6545 */
+        if (u < 1 * USEC_PER_MSEC || u > 1 * USEC_PER_SEC) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "CPU quota period '%s' outside of the valid range [1ms, 1s], ignoring.", rvalue);
```

Comment:
should pass 0 instead of r as 5th argument. (is not actually going to make a difference during runtime, since r is 0, but is semantically more correct)